### DISCUSSION TITLE
我有一个好的想法：应用于语言学习，发音纠正

### DISCUSSION THREAD

**Discussion by Pantyhose-X** [crimx/ext-saladict#1407] [Ideas]
你好：开发人员
 建议有.exe版，exe版调用Win10安装的TTS引擎，，你可以免去购买APi的费用
 你可以开发一个“监听”键盘工具，或者基于[nvda](https://github.com/nvaccess/nvda)
1. `输入空格` 后TTS朗读单词，`输入回车键`后朗读句子。
2. 应用于语言学习。发音纠正，这样可以纠正发音和加深记忆，
3. 母语不是英语。需要学习英语，市场非常大 

例子
https://www.youtube.com/watch?v=YPBn6D6pf78&feature=youtu.be&t=213
[![Watch the video](https://user-images.githubusercontent.com/42504385/122677974-e2341e00-d238-11eb-9a10-f7fd00ffb7d9.JPG)](https://www.youtube.com/watch?v=YPBn6D6pf78&t=213s)

你可以基于[nvda](https://github.com/nvaccess/nvda)

* 关闭屏幕阅读器关闭所有其他语音提示。
* 独立使用键盘阅读器，只朗读我在键盘上输入的内容
![image](https://user-images.githubusercontent.com/42504385/125297889-c4f7f880-e37b-11eb-9731-7e0a20a47545.png)
Android可以做成输入法，可以基于[OpenBoard](https://github.com/dslul/openboard)或[Simple Keyboard](https://github.com/rkkr/simple-keyboard)，内置mozilla text-to-speech或调用[谷歌文本转语音TTS apk](https://play.google.com/store/apps/details?id=com.google.android.tts&hl=en_US&gl=US)。

## 额外信息

开源 TTS
[Mozilla 的 Text-to-Speech](https://github.com/mozilla/TTS)
[coqui-ai TTS](https://github.com/coqui-ai/TTS)

**Comment by crimx** [comment]
感谢反馈！发音学习正好我也比较感兴趣（有录过的一些[作品](https://music.163.com/#/program?id=2057474080)）。我认为发音纠正的关键不在口说而在耳听，只要耳朵能分辨，口找到发音位置是迟早的事情。所以我觉得利用机器发音去纠音这个方法不太靠谱。软件合成的声音跟真人还是有差距，特别是在不熟悉一门语言发音的情况下，很可能会被机器发音带偏。

学习发音还是用真人语料比较好，最好是视频可以看到人在说话。

**Reply by lasdocas** [reply]
完全赞同。即便是电脑上的自然人声，“合成腔”还是非常明显。甚至从3年前开始，一些教科书（比如interchange或者touchstone）上配备的真人录音的听力练习，（作为一名教师）我都有减少使用，到现在几乎不用——因为即便那些练习是真人录音，但是还是有“录音腔”——语速和腔调与真实生活还是有差别。

**Reply by lasdocas** [reply]
*因为提到教科书，顺便补充一下，教科书也不用了 ><
